ast/src/object/mod.rs

```diff
@@ -24,4 +24,34 @@ pub struct Object<T> {
     pub degree: u64,
     /// the pil identities for this machine
     pub pil: Vec<PilStatement<T>>,
+    /// the links from this machine to its children
+    pub links: Vec<Link>,
+}
+
+pub struct Link {
+    pub from: LinkFrom,
+    pub to: LinkTo,
+}
+
+pub struct LinkFrom {
+    pub instr: Instr,
+}
+
+pub struct LinkTo {
+    pub machine_ty: String,
+    pub function: Function,
+    pub loc: String,
+    pub latch: Option<String>,
+}
+
+pub struct Instr {
+    pub flag: String,
```

Comment:
What's this flag?

Reply:
the instruction flag, currently `instr_foo` for `foo`. I just wanted to keep it explicit and not rely on pilgen. This translated directly into the lookup.